boards/imix/src/components/radio.rs

```diff
@@ -0,0 +1,119 @@
+// copies application transmissions into or copies out to application buffers
+// for reception.
+static mut RADIO_BUF: [u8; radio::MAX_BUF_SIZE] = [0x00; radio::MAX_BUF_SIZE];
+
+// The buffer RF233 packets are received into.
+static mut RF233_RX_BUF: [u8; radio::MAX_BUF_SIZE] = [0x00; radio::MAX_BUF_SIZE];
+
+// This buffer is used as an intermediate buffer for AES CCM encryption
+// An upper bound on the required size is 3 * BLOCK_SIZE + radio::MAX_BUF_SIZE
+const CRYPT_SIZE: usize = 3 * symmetric_encryption::AES128_BLOCK_SIZE + radio::MAX_BUF_SIZE;
+static mut CRYPT_BUF: [u8; CRYPT_SIZE] = [0x00; CRYPT_SIZE];
+
+
+
+
+impl Component for RadioComponent {
+    type Output = &'static capsules::ieee802154::RadioDriver<'static>;
+
+    unsafe fn finalize(&mut self) -> Self::Output {
+        let aes_ccm = static_init!(
+            capsules::aes_ccm::AES128CCM<'static, sam4l::aes::Aes<'static>>,
+            capsules::aes_ccm::AES128CCM::new(&sam4l::aes::AES, &mut CRYPT_BUF)
+        );
+        sam4l::aes::AES.set_client(aes_ccm);
+        sam4l::aes::AES.enable();
+
+        // Keeps the radio on permanently; pass-through layer
+        let awake_mac: &AwakeMac<RF233Device> =
+            static_init!(AwakeMac<'static, RF233Device>, AwakeMac::new(self.rf233));
+        self.rf233.set_transmit_client(awake_mac);
+        self.rf233.set_receive_client(awake_mac, &mut RF233_RX_BUF);
+
+        let mac_device = static_init!(
+            capsules::ieee802154::framer::Framer<
+                'static,
+            AwakeMac<'static, RF233Device>,
+            capsules::aes_ccm::AES128CCM<'static, sam4l::aes::Aes<'static>>,
+            >,
+            capsules::ieee802154::framer::Framer::new(awake_mac, aes_ccm)
+        );
+        aes_ccm.set_client(mac_device);
+        awake_mac.set_transmit_client(mac_device);
+        awake_mac.set_receive_client(mac_device);
+        awake_mac.set_config_client(mac_device);
+
+        let mux_mac = static_init!(
+            capsules::ieee802154::virtual_mac::MuxMac<'static>,
+            capsules::ieee802154::virtual_mac::MuxMac::new(mac_device)
+        );
+        mac_device.set_transmit_client(mux_mac);
+        mac_device.set_receive_client(mux_mac);
+
+        let radio_mac = static_init!(
+            capsules::ieee802154::virtual_mac::MacUser<'static>,
+            capsules::ieee802154::virtual_mac::MacUser::new(mux_mac)
+        );
+        mux_mac.add_user(radio_mac);
+
+        let radio_driver = static_init!(
+            capsules::ieee802154::RadioDriver<'static>,
+            capsules::ieee802154::RadioDriver::new(radio_mac, kernel::Grant::create(), &mut RADIO_BUF)
+        );
+
+        mac_device.set_key_procedure(radio_driver);
+        mac_device.set_device_procedure(radio_driver);
+        radio_mac.set_transmit_client(radio_driver);
+        radio_mac.set_receive_client(radio_driver);
+        radio_mac.set_pan(0xABCD);
```

Comment:
With this design, it is now pretty non-obvious where values such as the radio PAN and address should be set. When these values were set in main.rs it was likely that anyone using the stack would stumble across them pretty quickly, but if we are going to bury these constants in imix/src/components, it seems like we should at least add a comment directing developers here, or, alternatively, make the PAN and address parameters to finalize()

Reply:
I agree.

Reply:
They can't be arguments to finalize(), but can be arguments to new(). I updated components/radio.rs to do this. Now the boot sequence specifies the PanID and the short address.

One note -- the PanID is a type, but the short address isn't (it's just a u16). Might be useful to be consistent on this.